src/acquisition/covidcast/csv_importer.py

```diff
@@ -0,0 +1,242 @@
+"""Collects and reads covidcast data from a set of local CSV files."""
+
+# standard library
+import glob
+import os
+import re
+
+# third party
+import pandas
+
+
+class CsvImporter:
+  """Finds and parses covidcast CSV files."""
+
+  # .../source/yyyymmdd_geo_signal.csv
+  PATTERN_DAILY = re.compile(r'^.*/([^/]*)/(\d{8})_(\w+?)_(\w+)\.csv$')
+
+  # .../source/weekly_yyyyww_geo_signal.csv
+  PATTERN_WEEKLY = re.compile(r'^.*/([^/]*)/weekly_(\d{6})_(\w+?)_(\w+)\.csv$')
+
+  # set of allowed resolutions (aka "geo_type")
+  GEOGRAPHIC_RESOLUTIONS = {'county', 'hrr', 'msa', 'dma', 'state'}
+
+  # set of required CSV columns
+  REQUIRED_COLUMNS = {'geo_id', 'val', 'se', 'direction', 'sample_size'}
+
+  # NOTE: this should be a Python 3.7+ `dataclass`, but the server is on 3.4
+  # See https://docs.python.org/3/library/dataclasses.html
+  class RowValues:
+    """A container for the values of a single covidcast row."""
+
+    def __init__(self, geo_value, value, stderr, sample_size, direction):
+      self.geo_value = geo_value
+      self.value = value
+      self.stderr = stderr
+      self.sample_size = sample_size
+      self.direction = direction
+
+  @staticmethod
+  def is_sane_day(value):
+    """Return whether `value` is a sane (maybe not valid) YYYYMMDD date."""
+
+    year, month, day = value // 10000, (value % 10000) // 100, value % 100
+
+    nearby_year = 2014 <= year <= 2023
```

Comment:
why such a low value for the max year?

in either case, perhaps put these year boundary values as constants near the top of the file where they are more visible and easier to maintain if needed? similar for `is_sane_week()`.

Reply:
a couple of the data sources have historical data that extends back 5+ years. it wasn't clear to me how much historical data we'd want to support. but on second thought, i'm inclined to agree that 2014 is too early with respect to the pandemic. i'll bump that up to, say, 2019.

good feedback, i will extract out those constants.

thanks for looking :)

Reply:
done

Reply:
thanks! i hadn't thought of the historical data scenario and was more concerned about the max date and this code being useful beyond 2023, but with the constants it should be easy/safe to extend in either direction if the need arises in the future.

Reply:
oh i see, i misread your original comment. as you said, this should be easy to extend, but maybe it's better to just extend it now per the original comment. moved max year up to 2030.